File: front/src/ts/data/Status.ts
enum Status {
    FIRED = 'FIRED',
    RECOMMENDED_FOR_PROMOTION = "RECOMMENDED_FOR_PROMOTION",
    REGULAR = 'REGULAR',
    PROBATION = 'PROBATION'
}

export const statusToString = new Map<Status, string>()
statusToString.set(Status.FIRED, "Уволен")
statusToString.set(Status.RECOMMENDED_FOR_PROMOTION, "Рекомендован к повышению")
statusToString.set(Status.REGULAR, "Сотрудник")
statusToString.set(Status.PROBATION, "Испытательный срок")

export function stringToStatus(status: string) {
    switch (status.toUpperCase()) {
        case "FIRED":
            return Status.FIRED
        case "RECOMMENDED_FOR_PROMOTION":
            return Status.RECOMMENDED_FOR_PROMOTION
        case "REGULAR":
            return Status.REGULAR
        case "PROBATION":
            return Status.PROBATION
        default:
            return null
    }
}

export default Status

File: back/src/main/java/com/example/back/model/dto/Status.java
package com.example.back.model.dto;

public enum Status {
    FIRED,
    RECOMMENDED_FOR_PROMOTION,
    REGULAR,
    PROBATION;

    public static Status stringToStatus(String s) {
        return switch (s.toLowerCase()) {
            case "fired" -> Status.FIRED;
            case "recommended_for_promotion" -> Status.RECOMMENDED_FOR_PROMOTION;
            case "regular" -> Status.REGULAR;
            case "probation" -> Status.PROBATION;
            default -> null;
        };
    }
}